lang/cs/Org.Apache.REEF.Experimental/ParquetReader/ParquetReader.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Linq;
+using System.Diagnostics;
+using System.IO;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Experimental.ParquetReader.Parameters;
+using Org.Apache.REEF.Experimental.ParquetCollection;
+
+namespace Org.Apache.REEF.Experimental.ParquetReader
+{
+    /// <summary>
+    /// Constructs a Parquet file reader.
+    /// </summary>
+    public class ParquetReader : IDisposable
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(ParquetReader));
+
+        private bool _disposed = false;
+
+        private class JavaProcessConfiguration
+        {
+            public const string javaExecutable = "java";
+            public const string mainClass = "org.apache.reef.experimental.parquet.ParquetReader";
+            public string ClassPath { get; set; }
+        }
+
+        private class JavaProcess
```

Comment:
This class could be useful more broadly. Specifically, we solve similar problems in O.A.R.Client, where we also run an external Java Process. Similarly, in O.A.R.IO, we have a `IFileSystem` implementation that interfaces with an external Java process. Have a look at those and see whether you can reference their solution.